src/clr-angular/data/tree-view/tree-features.service.ts

```diff
@@ -8,6 +8,7 @@ import { Injectable, Optional, SkipSelf } from '@angular/core';
 @Injectable()
 export class TreeFeaturesService<T> {
   selectable = false;
+  smart = true;
```

Comment:
Calling this `smart` bothered me in Datagrid. We also use `clrLazy` to say it isn't `smart`, which requires a little bit of mental calculation. Do other features depend on this flag being set or could we `s/smart/lazy` and keep the correlation between the input and the service?

Reply:
As discussed, changing to `lazy` means every time we test it in our logic, it would be:
```typescript
if (!lazy) {
  // Propagate selection
}
```
```typescript
if (!lazy) {
  // Build children nodes
}
```
```typescript
if (!lazy) {
  // Add some enhanced feature on our side
}
```

We agreed this would force the "flip" gymnastic every time we right code, which is not desirable. To keep the property closer to the corresponding input, we agreed to rename it `eager`, which is more understandable.